docs/markdown/Contributions/releases/release-process.md

```diff
@@ -207,6 +207,8 @@ Run this script as a basic smoke test:
 
 You should also [check PyPI](https://pypi.org/pypi/pantsbuild.pants) to ensure everything looks good. Click "Release history" to find the version you released, then click it and confirm the changelog is correct on the "Project description" page and that the `macOS` and `manylinux` wheels show up in the "Download files" page. 
```

Comment:
NB. PyPI is still used in 2.17, so I left this paragraph in intentionally, unlike #19924 that fully replaced it.